contiamo/utils.py

```diff
@@ -54,47 +54,74 @@ def contract_url_template_from_identifier(contract_identifier, api_base):
 
 ###
 # parse query response
-def raise_response_error(e, response, logger=None):
+def raise_json_error(e, response):
   from contiamo.errors import ResponseError
-  if logger:
-    logger.error('Invalid JSON response: %s' % response.text)
   raise ResponseError(
     'The JSON response from the server was invalid. Please report the bug to support@contiamo.com\n'
     'The following %s error was raised when parsing the JSON response:\n%s' % (type(e).__name__, e),
-    http_body=response.content, http_status=response.status_code, headers=response.headers)
+    json_body=response)
 
 def parse_query_result(json_response, parse_dates=True, use_column_names=True):
   if not pd:
     return json_response
 
+  # This function is not safe (only works with floats) so we keep it in the local namespace.
+  def is_integer(a):
+    return a.is_integer()
+
   df = pd.DataFrame()
   try:
-    for idx, c in enumerate(json_response['columns']):
-      if use_column_names:
-        col_name = c.get('name', 'Date')
-      else:
-        col_name = c['key']
+    # parse column names and handle duplicates
+    if use_column_names:
+      column_names = []
+      for column in json_response['columns']:
+        if 'name' in column:
+          column_names.append(column['name'])
+        else:
+          # column is a date, but can be the comparison date
+          if column.get('comparison'):
+            column_names.append('Date (comparison)')
+          else:
+            column_names.append('Date')
+      if len(column_names) > len(set(column_names)):
+        # column names not unique: use app_data name for metrics
+        for idx, column in enumerate(json_response['columns']):
+          if column['column_type'] == 'metric':
+            column_names[idx] = column['app_data']['name'] + ': ' + column['name']
```

Comment:
I only include the app data name for metrics, so that should work, right? But we could be in the situation where column_names are not unique because of a calculation, in which case adding app names to metrics wouldn't help. I could look for duplicates among metrics only...
There has always been a flag to use column keys rather than names.